src/install_pypi/mod.rs

```diff
@@ -223,54 +185,226 @@ impl<'a> PyPIPrefixUpdater<'a> {
-            .plan_install(&site_packages, registry_index, &required_map)
-            .into_diagnostic()
-            .context("error while determining PyPI installation plan")?;
-
-        // Determine the currently installed conda packages.
-        let installed_packages = self.prefix.find_installed_packages().with_context(|| {
-            format!(
-                "failed to determine the currently installed packages for {}",
-                self.prefix.root().display()
-            )
-        })?;
-
-        let pypi_conda_clobber = PypiCondaClobberRegistry::with_conda_packages(&installed_packages);
+        let installation_plan =
+            InstallPlanner::new(self.uv_context.cache.clone(), &self.lock_file_dir)
+                .plan(&site_packages, registry_index, &required_map)
+                .into_diagnostic()
+                .context("error while determining PyPI installation plan")?;
 
         // Show totals
-        let total_to_install = local.len() + remote.len();
+        let total_to_install = installation_plan.local.len() + installation_plan.remote.len();
         let total_required = required_map.len();
         tracing::info!(
             "{} of {} required packages are considered are installed and up-to-date",
             total_required - total_to_install,
             total_required
         );
+
+        // Create the updater
+        let updater = PyPIPrefixUpdater {
+            prefix: self.prefix,
+            venv: self.venv,
+            uv_context: self.uv_context.clone(),
+            registry_client: self.registry_client,
+            environment_variables: self.environment_variables,
+            tags: self.tags,
+            flat_index: self.flat_index,
+            config_settings: self.config_settings,
+            build_options: self.build_options,
+            index_locations: self.index_locations,
+            build_isolation: self.build_isolation,
+            installation_plan,
+        };
+
+        Ok(updater)
+    }
+}
+
+/// This installs a PyPI prefix given a specific installation plan.
+pub struct PyPIPrefixUpdater {
+    prefix: Prefix,
+    venv: PythonEnvironment,
+    uv_context: UvResolutionContext,
+    registry_client: Arc<RegistryClient>,
+    environment_variables: HashMap<String, String>,
+    tags: Tags,
+    flat_index: FlatIndex,
+    config_settings: ConfigSettings,
+    build_options: BuildOptions,
+    index_locations: IndexLocations,
+    build_isolation: BuildIsolation,
+    installation_plan: PyPIInstallationPlan,
+}
+
+impl PyPIPrefixUpdater {
+    /// Remove metadata for duplicate packages
+    fn remove_duplicate_metadata(&self, duplicates: &[InstalledDist]) -> std::io::Result<()> {
+        for duplicate in duplicates {
+            tracing::debug!(
+                "Removing metadata for duplicate package: {}",
+                duplicate.name()
+            );
+            fs_err::remove_dir_all(duplicate.install_path())?;
+        }
+        Ok(())
+    }
```

Comment:
Anything that is extraneous, but considered a duplicate we just remove the `dist-info`.

Example
```
site-packages/
├── foobar/ (actual package)
├── foobar-0.1.0.dist-info/
│   └── INSTALLER (Conda)l
└── foobar-0.1.1.dist-info/
    └── INSTALLER (PyPI)
```

We only remove the 0.1.1 dist-info and leave the rest alone.